server/services/componentNameService.ts

```diff
@@ -103,4 +103,12 @@ export default class ComponentNameService {
     const components = componentData.find(repoName => formatMonitorName(repoName.github_repo) === repositoryName)
     return !!components
   }
+
+  async checkComponentArchiveRequestExists(repositoryName: string): Promise<boolean> {
+    const componentData = await this.strapiApiClientFactory('').getGithubRepoRequests()
+    const components = componentData.find(
```

Comment:
Maybe could use `.some` and then just return here instead of `return !!components` as you don't care about the index?